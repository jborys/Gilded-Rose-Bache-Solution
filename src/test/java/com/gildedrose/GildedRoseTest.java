package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static com.gildedrose.GildedRoseConstants.*;

class GildedRoseTest {

    @Test
    void foo() throws Exception {
        CombinationApprovals.verifyAllCombinations(
            this::performUpdateQuality,
            new String[] {"foo", AGED_BRIE, BACKSTAGE_PASSES, HAND_OF_RAGNAROS},
            new Integer[]{-1, 0, 5, 10, 11 },
            new Integer[]{0, 1, 49, 55}
        );
    }

    private Item performUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] {Item.createItem(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        Item item = app.items[0];
        app.updateQuality();
        return item;
    }

}
