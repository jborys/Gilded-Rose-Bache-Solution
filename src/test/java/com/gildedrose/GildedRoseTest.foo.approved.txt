[foo, -1, 0] => Item{name='foo', sellIn=-2, quality=0} 
[foo, -1, 1] => Item{name='foo', sellIn=-2, quality=0} 
[foo, -1, 49] => Item{name='foo', sellIn=-2, quality=47} 
[foo, -1, 55] => Item{name='foo', sellIn=-2, quality=53} 
[foo, 0, 0] => Item{name='foo', sellIn=-1, quality=0} 
[foo, 0, 1] => Item{name='foo', sellIn=-1, quality=0} 
[foo, 0, 49] => Item{name='foo', sellIn=-1, quality=47} 
[foo, 0, 55] => Item{name='foo', sellIn=-1, quality=53} 
[foo, 5, 0] => Item{name='foo', sellIn=4, quality=0} 
[foo, 5, 1] => Item{name='foo', sellIn=4, quality=0} 
[foo, 5, 49] => Item{name='foo', sellIn=4, quality=48} 
[foo, 5, 55] => Item{name='foo', sellIn=4, quality=54} 
[foo, 10, 0] => Item{name='foo', sellIn=9, quality=0} 
[foo, 10, 1] => Item{name='foo', sellIn=9, quality=0} 
[foo, 10, 49] => Item{name='foo', sellIn=9, quality=48} 
[foo, 10, 55] => Item{name='foo', sellIn=9, quality=54} 
[foo, 11, 0] => Item{name='foo', sellIn=10, quality=0} 
[foo, 11, 1] => Item{name='foo', sellIn=10, quality=0} 
[foo, 11, 49] => Item{name='foo', sellIn=10, quality=48} 
[foo, 11, 55] => Item{name='foo', sellIn=10, quality=54} 
[Aged Brie, -1, 0] => Item{name='Aged Brie', sellIn=-2, quality=2} 
[Aged Brie, -1, 1] => Item{name='Aged Brie', sellIn=-2, quality=3} 
[Aged Brie, -1, 49] => Item{name='Aged Brie', sellIn=-2, quality=50} 
[Aged Brie, -1, 55] => Item{name='Aged Brie', sellIn=-2, quality=55} 
[Aged Brie, 0, 0] => Item{name='Aged Brie', sellIn=-1, quality=2} 
[Aged Brie, 0, 1] => Item{name='Aged Brie', sellIn=-1, quality=3} 
[Aged Brie, 0, 49] => Item{name='Aged Brie', sellIn=-1, quality=50} 
[Aged Brie, 0, 55] => Item{name='Aged Brie', sellIn=-1, quality=55} 
[Aged Brie, 5, 0] => Item{name='Aged Brie', sellIn=4, quality=1} 
[Aged Brie, 5, 1] => Item{name='Aged Brie', sellIn=4, quality=2} 
[Aged Brie, 5, 49] => Item{name='Aged Brie', sellIn=4, quality=50} 
[Aged Brie, 5, 55] => Item{name='Aged Brie', sellIn=4, quality=55} 
[Aged Brie, 10, 0] => Item{name='Aged Brie', sellIn=9, quality=1} 
[Aged Brie, 10, 1] => Item{name='Aged Brie', sellIn=9, quality=2} 
[Aged Brie, 10, 49] => Item{name='Aged Brie', sellIn=9, quality=50} 
[Aged Brie, 10, 55] => Item{name='Aged Brie', sellIn=9, quality=55} 
[Aged Brie, 11, 0] => Item{name='Aged Brie', sellIn=10, quality=1} 
[Aged Brie, 11, 1] => Item{name='Aged Brie', sellIn=10, quality=2} 
[Aged Brie, 11, 49] => Item{name='Aged Brie', sellIn=10, quality=50} 
[Aged Brie, 11, 55] => Item{name='Aged Brie', sellIn=10, quality=55} 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, -1, 55] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 0, 55] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0} 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=4, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=4, quality=4} 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=4, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 5, 55] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=4, quality=55} 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=9, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=9, quality=3} 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=9, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 10, 55] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=9, quality=55} 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=1} 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=2} 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=50} 
[Backstage passes to a TAFKAL80ETC concert, 11, 55] => Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=55} 
[Sulfuras, Hand of Ragnaros, -1, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=0} 
[Sulfuras, Hand of Ragnaros, -1, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=1} 
[Sulfuras, Hand of Ragnaros, -1, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=49} 
[Sulfuras, Hand of Ragnaros, -1, 55] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=55} 
[Sulfuras, Hand of Ragnaros, 0, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=0} 
[Sulfuras, Hand of Ragnaros, 0, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=1} 
[Sulfuras, Hand of Ragnaros, 0, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=49} 
[Sulfuras, Hand of Ragnaros, 0, 55] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=55} 
[Sulfuras, Hand of Ragnaros, 5, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=5, quality=0} 
[Sulfuras, Hand of Ragnaros, 5, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=5, quality=1} 
[Sulfuras, Hand of Ragnaros, 5, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=5, quality=49} 
[Sulfuras, Hand of Ragnaros, 5, 55] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=5, quality=55} 
[Sulfuras, Hand of Ragnaros, 10, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=10, quality=0} 
[Sulfuras, Hand of Ragnaros, 10, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=10, quality=1} 
[Sulfuras, Hand of Ragnaros, 10, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=10, quality=49} 
[Sulfuras, Hand of Ragnaros, 10, 55] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=10, quality=55} 
[Sulfuras, Hand of Ragnaros, 11, 0] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=0} 
[Sulfuras, Hand of Ragnaros, 11, 1] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=1} 
[Sulfuras, Hand of Ragnaros, 11, 49] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=49} 
[Sulfuras, Hand of Ragnaros, 11, 55] => Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=55} 
